file.go

```diff
@@ -0,0 +1,100 @@
+package sivafs
+
+import (
+	"io"
+	"sync"
+
+	"gopkg.in/src-d/go-billy.v2"
+	"gopkg.in/src-d/go-siva.v1"
+)
+
+type file struct {
+	billy.BaseFile
+
+	w siva.Writer
+	r *io.SectionReader
+	l *sync.Mutex
+
+	closeNotify func() error
+}
+
+func newFile(filename string, w siva.Writer,
+	l *sync.Mutex, closeNotify func() error) billy.File {
```

Comment:
The idea here is to have a main lock, because is not possible do more than one operation at the time using siva (we cannot open two files and do several operations on both).